components/eventing-controller/pkg/backend/jetstreamv2/test_helpers.go

```diff
@@ -0,0 +1,94 @@
+	natstesting "github.com/kyma-project/kyma/components/eventing-controller/pkg/backend/nats/testing"
+	"github.com/kyma-project/kyma/components/eventing-controller/pkg/ems/api/events/types"
+	evtesting "github.com/kyma-project/kyma/components/eventing-controller/testing"
+	evtestingv2 "github.com/kyma-project/kyma/components/eventing-controller/testing/v2"
+)
+
+func SendEventToJetStream(jsClient *JetStream, data string) error {
+	// assumption: the event-type used for publishing is already cleaned from none-alphanumeric characters
+	// because the publisher-application should have cleaned it already before publishing
+	eventType := evtestingv2.OrderCreatedCleanEvent
+	eventTime := time.Now().Format(time.RFC3339)
+	sampleEvent := natstesting.NewNatsMessagePayload(data, "id", evtestingv2.EventSourceClean, eventTime, eventType)
+	return jsClient.Conn.Publish(jsClient.getJetStreamSubject(evtestingv2.EventSourceClean, eventType, v1alpha2.TypeMatchingStandard), []byte(sampleEvent))
+}
+
+func sendEventToJetStreamOnEventType(jsClient *JetStream, eventType string, data string, typeMatching v1alpha2.TypeMatching) error {
+	eventTime := time.Now().Format(time.RFC3339)
+	sampleEvent := natstesting.NewNatsMessagePayload(data, "id", evtestingv2.EventSourceClean, eventTime, eventType)
+	return jsClient.Conn.Publish(jsClient.getJetStreamSubject(evtestingv2.EventSourceClean, eventType, typeMatching), []byte(sampleEvent))
+}
+
+func sendCloudEventToJetStream(jetStreamClient *JetStream, subject, eventData, cetype string) error {
+	// create a CE http request
+	var headers http.Header
+	if cetype == types.ContentModeBinary {
+		headers = evtesting.GetBinaryMessageHeaders()
+	} else {
+		headers = evtesting.GetStructuredMessageHeaders()
+	}
+	req, err := http.NewRequest(http.MethodPost, "dummy", bytes.NewBuffer([]byte(eventData)))
+	if err != nil {
+		return err
+	}
+	for k, v := range headers {
+		req.Header[k] = v
+	}
+	// convert  to the CE Event
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	message := cev2http.NewMessageFromHttpRequest(req)
+	defer func() { _ = message.Finish(nil) }()
+	event, err := binding.ToEvent(ctx, message)
+	if err != nil {
+		return err
+	}
+	if err := event.Validate(); err != nil {
+		return err
+	}
+	// get a CE sender for the embedded NATS using CE-SDK
+	natsOpts := nats2.NatsOptions()
+	url := jetStreamClient.Config.URL
+	sender, err := nats2.NewSender(url, subject, natsOpts)
+	if err != nil {
+		return nil
+	}
+	client, err := v2.NewClient(sender)
+	if err != nil {
+		return err
+	}
+	// force binary binding and send the event to NATS using CE-SDK
+	if cetype == types.ContentModeBinary {
+		ctx = binding.WithForceBinary(ctx)
+	} else {
+		ctx = binding.WithForceStructured(ctx)
+	}
+	if err := client.Send(ctx, *event); err != nil {
+		return err
+	}
+	return nil
+}
+
+func AddJSCleanEventTypesToStatus(sub *v1alpha2.Subscription, cleaner cleaner.Cleaner) error {
```

Comment:
Should we move this file to testing pkg? The methods seems useful for reconciler testing as well!

Reply:
Actually this was inside the testing file. I just exported it since I needed it in the shared tests for v1 and v2. I don't think its needed in the reconciler tests. If I do, I'll move it later